src/cycax/cycad/engines/simple_2d.py

```diff
@@ -140,19 +140,19 @@ def _figure_feature(self, ax: matplotlib.axes._axes.Axes, feature: dict):
         elif feature_type == "hole":
             self._hole(ax, feature)
 
-    def save_as_figure(self):
+    def build(self):
         """This method will coordinate the drawing and saving of the figure that is being decoded from the provided json."""
-        in_name = os.getcwd() + "/" + self.part_no + "/" + self.part_no + ".json"
+        in_name = "{cwd}/{name}/{name}.json".format(cwd=os.getcwd(), name=self.name)
```

Comment:
When a child class of PartEngine use self._json_file to get a Path object to the file.